ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,146 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Upload commands.
+"""
+
+import os
+from contextlib import closing
+from collections import namedtuple
+
+from . import _api
+from . import _options
+from . import _ovirt
+from . import _ui
+
+
+DiskInfo = namedtuple(
+    "DiskInfo",
+    "name,initial_size,provisioned_size,content_type,format")
+
+
+def register(parser):
+    cmd = parser.add_sub_command(
+        "upload-disk",
+        help="Upload disk",
+        func=upload_disk)
+
+    cmd.add_argument(
+        "-s", "--storage-domain",
+        required=True,
+        help="Name of the storage domain.")
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
```

Comment:
Should this actually default to qcow2? If I as a user am uploading a raw image and I do not set the format, I would expect it to keep the uploaded disk raw (i.e., if not set, respect the source format).

Reply:
I would expect that too, as a user, that the default is "whatever the image is now".

Reply:
If we keep the default value as is, there is this assignation below that is not needed:
```python
    format = dst_format or img_info["format"]
```
This will always have a `dst_format`.

Otherwise we can default the argument to None and keep the line

Reply:
Uploading the same format is a naive expectation. The system has good reason to use specific format for specific cases and defaulting to the same format the client has is not a useful default.

The typical images users may upload are:
- appliance, typically qcow2 compressed (best format for publishing images)
- backup when restoring vm from backup, typically qcow2 (download default)
- disk from another system, typically qcow2 (download default)

When uploading we want to ensure:
- Upload using defaults just works with any storage type
- Disk can be used with incremental backup
- Disk use thin provisioning to reduce resources usage

Using qcow2 sparse images works on any storage, can be used for incremental backup and minimize storage space used on the system - so this is a good default.

Reply:
Ok, great clarification :) Then we keep it as is.